### PR TITLE
Add support for arbitrary HTTP request options #4

### PR DESCRIPTION
**Pull request by eranharel**

### CODE REVIEW

File: checks/http.go [hunk 1]
```diff
@@ -33,8 +33,13 @@ type HTTPCheckConfig struct {
 	Client *http.Client
 	// Timeout is the timeout used for the HTTP request, defaults to "1s".
 	Timeout time.Duration
+	// Options allow you to configure the HTTP request with arbitrary settings, e.g. add request headers, etc.
+	Options []RequestOption
 }
 
+// RequestOption configures the request with arbitrary settings, e.g. add request headers, etc.
```
Comment (rantav) [inline checks/http.go]: Isn't this too permissive? If you get a request at your hand you can configure it, but you can do more. E.g. execute it right?...
I'm not sure (didn't check) but is there a way to make it less permissive?

File: checks/http_test.go [hunk 2]
```diff
@@ -5,17 +5,48 @@ import (
...
+func (req *receivedRequest) getDetail(key string) string {
+	req.RLock()
+	defer req.RUnlock()
+	return req.details[key]
+}
+
+func (req *receivedRequest) clear() {
+	req.Lock()
+	defer req.Unlock()
+	req.details = make(map[string]string, 2)
```
Comment (rantav) [inline checks/http_test.go]: why `2`?

Comment (eranharel) [reply checks/http_test.go]: It's just a test... I make up to 2 changes in the test that uses that...

File: checks/http.go (hunk 1 above)
Comment (eranharel) [reply checks/http.go]: Request is a struct, and people need to access req.Header to modify the headers, so I don't think I can extract a restrictive interface unless I totally wrap the request or let the user create one from scratch. Got any concrete ideas here?

Comment (rantav) [reply checks/http.go]: You could wrap is, but that's a lot of work, nevermind

**Pull request merged**